components/datadog/apps/tracegen/k8s.go

```diff
@@ -0,0 +1,168 @@
+package tracegen
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+
+	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes"
+	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/apps/v1"
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type K8sComponent struct {
+	pulumi.ResourceState
+}
+
+func K8sAppDefinition(e config.CommonEnvironment, kubeProvider *kubernetes.Provider, namespace string, traceAgentURL string, nodeGroup pulumi.StringInput, opts ...pulumi.ResourceOption) (*K8sComponent, error) {
+	opts = append(opts, pulumi.Provider(kubeProvider), pulumi.Parent(kubeProvider), pulumi.DeletedWith(kubeProvider))
+
+	k8sComponent := &K8sComponent{}
+	if err := e.Ctx.RegisterComponentResource("dd:apps", fmt.Sprintf("%s/tracegen", namespace), k8sComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(k8sComponent))
+
+	ns, err := corev1.NewNamespace(e.Ctx, namespace, &corev1.NamespaceArgs{
+		Metadata: metav1.ObjectMetaArgs{
+			Name: pulumi.String(namespace),
+		},
+	}, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, utils.PulumiDependsOn(ns))
+
+	if _, err := appsv1.NewDeployment(e.Ctx, fmt.Sprintf("%s/tracegen-uds", namespace), &appsv1.DeploymentArgs{
+		Metadata: &metav1.ObjectMetaArgs{
+			Name:      pulumi.String("tracegen-uds"),
+			Namespace: pulumi.String(namespace),
+			Labels: pulumi.StringMap{
+				"app": pulumi.String("tracegen-uds"),
+			},
+		},
+		Spec: &appsv1.DeploymentSpecArgs{
+			Replicas: pulumi.Int(1),
+			Selector: &metav1.LabelSelectorArgs{
+				MatchLabels: pulumi.StringMap{
+					"app": pulumi.String("tracegen-uds"),
+				},
+			},
+			Template: &corev1.PodTemplateSpecArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("tracegen-uds"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					NodeSelector: pulumi.StringMap{
+						"eks.amazonaws.com/nodegroup": nodeGroup,
+					},
+					Containers: corev1.ContainerArray{
+						&corev1.ContainerArgs{
+							Name:  pulumi.String("tracegen-uds"),
+							Image: pulumi.String("ghcr.io/datadog/apps-tracegen:main"),
+							Env: &corev1.EnvVarArray{
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_TRACE_AGENT_URL"),
+									Value: pulumi.String(traceAgentURL),
```

Comment:
I would be fine with a hard-code of the value here.

```suggestion
									Value: pulumi.String("/var/run/datadog/apm.socket"),
```

Having a `traceAgentURL` parameter in the `K8sAppDefinition` function gives the feeling that we can change the path of the `trace-agent` socket.
But in practice, it’s false. If we pass `/run/apm.socket` for ex., it won’t work because the `apmsocketpath` `HostPath` is hardcoded to `/var/run/datadog` line 98.

So, we would need to make `HostPath` line 98 use the dirname of `traceAgentURL`.
This sounds overkill to me.
That’s why I’d rather hardcode the socket path here and remove the `traceAgentURL` parameter from the `K8sAppDefinition`.

Also, it looked weird to be able to customize the UDS path in UDS case, but not the TCP port in the TCP case.

Reply:
I hardcode the path in the last commit 👍